solr/core/src/java/org/apache/solr/update/UpdateHandler.java

```diff
@@ -200,4 +187,16 @@ public void registerOptimizeCallback( SolrEventListener listener )
   }
 
   public abstract void split(SplitIndexCommand cmd) throws IOException;
+  
+  private static UpdateLog initialisePluginUpdateLog(SolrCore core, String dataDir, PluginInfo ulogPluginInfo)
+  {
+    String className = ulogPluginInfo.className;
+    if (System.getProperty("test.hdfs.forceHdfsUpdateLog") != null) {
+      className = "solr.HdfsUpdateLog";
+    }
+    if (className != null) {
+      return core.getResourceLoader().newInstance(className, UpdateLog.class);
+    }
+    return new UpdateLog();
+  }
 }
```

Comment:
I wonder if a nicer way of doing this would be to make UpdateLog be created by the DirectoryFactory? Something like:

```
DirectoryFactory { 
  public UpdateLog newUpdateLog(SolrCore core, String dataDir, PluginInfo ulogPluginInfo) {
    if (ulogPluginInfo.className == null)
      return new UpdateLog();
    return core.getResourceLoader().newInstance(ulogPluginInfo.className, UpdateLog.class);
  }
}

HdfsDirectoryFactory {
  @Override
  public UpdateLog newUpdateLog( ... ) {
    if (ulogPluginInfo.className == null) {
      return new HdfsUpdateLog();
    }
    return super.newUpdateLog( ... );
}
```